Index: b/sim/arm/maverick.c
===================================================================
--- a/sim/arm/maverick.c
+++ b/sim/arm/maverick.c
@@ -61,7 +61,7 @@ union maverick_acc_regs
 };
 
 static struct maverick_regs DSPregs[16];
-union maverick_acc_regs DSPacc[4];
+static union maverick_acc_regs DSPacc[4];
 ARMword DSPsc;
 
 #define DEST_REG	(BITS (12, 15))
Index: b/sim/arm/wrapper.c
===================================================================
--- a/sim/arm/wrapper.c
+++ b/sim/arm/wrapper.c
@@ -130,7 +130,7 @@ union maverick_acc_regs
 };
 
 static struct maverick_regs     DSPregs[16];
-union maverick_acc_regs  DSPacc[4];
+static union maverick_acc_regs  DSPacc[4];
 ARMword DSPsc;
 
 static void
