Index: b/gprofng/src/Makefile.am
===================================================================
--- a/gprofng/src/Makefile.am
+++ b/gprofng/src/Makefile.am
@@ -179,10 +179,4 @@ $(srcdir)/DbeSession.cc: QLParser.tab.hh
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
-	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
-	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
-	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
-	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
-	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
Index: b/gprofng/src/Makefile.in
===================================================================
--- a/gprofng/src/Makefile.in
+++ b/gprofng/src/Makefile.in
@@ -1119,12 +1119,6 @@ $(srcdir)/DbeSession.cc: QLParser.tab.hh
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
-	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
-	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
-	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
-	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
-	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
