Index: glibc-2.14/configure
===================================================================
--- glibc-2.14.orig/configure	2011-05-31 12:12:33.000000000 +0800
+++ glibc-2.14/configure	2025-02-03 13:31:31.768127896 +0800
@@ -5002,7 +5002,7 @@
   ac_prog_version=`$MAKE --version 2>&1 | sed -n 's/^.*GNU Make[^0-9]*\([0-9][0-9.]*\).*$/\1/p'`
   case $ac_prog_version in
     '') ac_prog_version="v. ?.??, bad"; ac_verc_fail=yes;;
-    3.79* | 3.[89]*)
+    3.79* | 3.[89]* | 4.[0-9]*)
        ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
